code/game/objects/structures/floodlight.dm

```diff
@@ -0,0 +1,320 @@
+/obj/machinery/power/floodlight
+	name = "floodlight"
+	icon = 'icons/obj/floodlight.dmi'
+	icon_state = "floodlight0_hatchopen0"
+	desc = "An industrial floodlight. It has units of power remaining."
+	density = 1
+	anchored = 0
+	var/status = 0
+	var/obj/item/weapon/stock_parts/cell/powerpack
+	var/datum/effect_system/spark_spread/sparks = new
+	var/health = 60
+	var/maxhealth = 60
+	var/broken = 0
+	var/cover = 0
+	var/hascell = 0
```

Comment:
I still think it's unnecessary. You can replace instances of "hascell" with "powerpack" like so:
`hascell == 1` becomes `powerpack`
`hascell == 0` becomes `!powerpack`
`"floodlight[status]_hatchopen[hascell]"` becomes `"floodlight[status]_hatchopen[powerpack ? 1 : 0]"`